frontend/src/employee-frontend/components/UnitPage.tsx

```diff
@@ -148,6 +154,11 @@ export default React.memo(function UnitPage() {
               />
             )}
           />
+          <RouteWithTitle
+            exact
+            path="/units/:id/calendar"
+            component={() => <TabCalendar unitId={id} />}
```

Comment:
This arrow function is different every time `UnitPage` is rerendered and thus it's a different component each time. In practice this mounts and unmounts `TabCalendar` as many times as `UnitPage` is rerendered. This means for example that the data used by the component is requested 5 times. To prevent that either use `Route`'s `render` or `children` props or just get the `unitId` prop from path using `useParam`.